routes/repository.js

```diff
@@ -87,26 +87,22 @@ router.post("/enable", authMiddleware.isLoggedIn, function(req, res, next) {
 router.post('/mail/disable', authMiddleware.isLoggedIn, function (req, res, next) {
   Repository.updateRepository({name: req.body.repository}, {'mailService.status': false})
   .then(function () {
-    res.redirect('/' + req.body.repository + '/settings?status=mail_disabled_successful');
+    res.json({success: true});
   })
   .catch(function () {
-    next({
-      status: 500,
-      messages: 'Something went wrong'
-    });
+    res.status(500);
+    res.json({success: false});
```

Comment:
When it was `next()`, a response was never received at the client side. Hence I had to make this change